dds/README.md

```diff
@@ -34,6 +34,10 @@ A basic example on how to use Dust DDS. The publisher side can be implemented as
             .create_participant(domain_id, QosKind::Default, NoOpListener::new(), NO_STATUS)
             .unwrap();
 
+        participant

```

Comment:
I still don't see the reason why this shall not me made more like in the standard. Something like:
```
trait TypeSupport {
    fn register_type(
        participant: &dust_dds::domain::domain_participant::DomainParticipant,
        type_name: &str,
    ) -> DdsResult<()> where Self: Sized + DdsKey + dust_dds::topic_definition::type_support::DdsHasKey + Send + Sync + 'static{
        participant.register_type(type_name, FooTypeSupport::<Self>::new())
    }
}
impl TypeSupport for UserType{}
```
Which resembles 2.2.2.3.6 TypeSupport Interface

And then calling: 
`UserType::register_type(&participant, "UserType").unwrap();`

There is also a get_type_name function which could be used to avoid the need of writing the type names as Strings twice